galasa-managers-parent/galasa-managers-cicsts-parent/dev.galasa.cicsts.manager.ivt/src/main/java/dev/galasa/cicsts/manager/ivt/CICSTSManagerIVT.java

```diff
@@ -0,0 +1,199 @@
+import dev.galasa.zosbatch.IZosBatchJobname;
+
+import org.apache.commons.logging.Log;
+
+@Test
+public class CICSTSManagerIVT {
+	
+   @Logger
+   public Log logger;
+
+   @CicsRegion
+   public ICicsRegion cics;
+
+   @CicsTerminal
+   public ICicsTerminal terminal;
+   
+   @ZosBatch(imageTag = "PRIMARY")
+   public IZosBatch batch;
+   
+   @ZosBatchJobname(imageTag = "PRIMARY")
+   public IZosBatchJobname jobName;
+   
+   @BeforeClass
+   public void login() throws KeyboardLockedException, NetworkException, TerminalInterruptedException, TimeoutException, FieldNotFoundException {
+      terminal.clear();
+      terminal.waitForKeyboard();
+   }
+	
+   @Test
+   public void testNotNull() {
+	   assertThat(logger).isNotNull();
+	   assertThat(cics).isNotNull();
+	   assertThat(terminal).isNotNull();
+   }
+   
+   /**
+    * Tests that the CICS TS Manager retrieves the correct APPLID for the CICS Region
+    * @throws FieldNotFoundException 
+    * @throws NetworkException 
+    * @throws TerminalInterruptedException 
+    * @throws KeyboardLockedException 
+    * @throws TimeoutException 
+    */
+   @Test 
+   public void testGetApplid() throws TimeoutException, KeyboardLockedException, TerminalInterruptedException, NetworkException, FieldNotFoundException  {
+	   logger.info("Testing that the CICS TS Manager gets the correct APPLID for the CICS Region");
+	   String testApplid = cics.getApplid();
+	   logger.info("APPLID retrieved by the CICS TS Manager: " + testApplid);
+	   terminal.type("CEMT INQUIRE").enter().waitForKeyboard();
+	   String terminalScreen = terminal.retrieveScreen();
+	   assertThat(terminalScreen.contains(testApplid)).isTrue();
+   }
+   
+   /**
+    * Tests that the CICS TS Manager retrieves a CICS Resource
+    * @throws CicstsManagerException 
+    */
+   @Test
+   public void testCicsResource() throws CicstsManagerException  {
+	   logger.info("Testing that the CICS TS Manager retrieves a CICS Resource");
+	   ICicsResource cicsResource = cics.cicsResource();
+	   assertThat(cicsResource).isNotNull();
+   }
+   
+   /**
+    * Tests that the CICS TS Manager retrieves the correct job
+    * @throws ZosBatchException 
+    * @throws FieldNotFoundException 
+    * @throws NetworkException 
+    * @throws TerminalInterruptedException 
+    * @throws KeyboardLockedException 
+    * @throws TimeoutException 
+    */
+   @Test
+   public void testGetRegionJob() throws CicstsManagerException, ZosBatchException, TimeoutException, KeyboardLockedException, TerminalInterruptedException, NetworkException, FieldNotFoundException {
+	   logger.info("Testing that the CICS TS Manager finds the DSE CICS job for the CICS Region");
+	   logger.info("There should be an active job as the CICS Region is running");
+	   boolean exceptionThrown = false;
+	   IZosBatchJob regionJob = null;
+	   try {
+		   regionJob = cics.getRegionJob();
+	   } catch (CicstsManagerException e) {
+		   exceptionThrown = true;
+	   }
```

Comment:
For clarity the method could declare the ManagerException and you could log the fact that you are about to obtain the regionJob.  if the method does throw the exception then it will be clear what has happened

Reply:
I have removed the try catch block and added logging to make it clearer what the test is attempting to do. If an exception is thrown, it is obvious that whatever we were trying to do failed.